### PR TITLE
[IRRC-19 & IRRC-83] Create matching table

### PR DESCRIPTION
**Pull request by dzldebby**

### CODE REVIEW

**Review by nondescryptid — CHANGES_REQUESTED** [review]
LGTM, just needs a minor change before merging - the "matched selected" button should be "confirm selected" (see Figma)! let me know when the change has been committed so i can approve the PR :D

File: frontend/src/pages/Matching.vue [hunk 1]
```diff
@@ -1,40 +1,480 @@
...
+                >
+                  <span>Unmatch Selected </span> ({{ checkedRows.length }})
+                </button>
+                <div class="divider"/>
+                <button
+                  class="button field is-white"
+                  @click="cardModal()"
+                  :disabled="!checkedRows.length"
+                >
+                  <span>Matched selected </span> ({{ checkedRows.length }})
```
Comment (nondescryptid) [inline frontend/src/pages/Matching.vue]: should be "confirm selected" as the button leads to a pop-up saying that matches will be confirmed + an email will be sent to teachers

**Review by nondescryptid — APPROVED** [review]

File: backend/src/routes/statusUpdates.js [hunk 2]
```diff
@@ -44,6 +44,8 @@ router.post('/', async (req, res) => {
     })
   }
 
+  const result = await statusUpdates.addStatusUpdate(req.body)
+
```
Comment (therealyingtong) [inline backend/src/routes/statusUpdates.js]:
```suggestion
```
`result` is declared on line 66.

File: frontend/src/store.js [hunk 3]
```diff
@@ -24,6 +24,10 @@ import {
   studentState
 } from "./store/students.js";
 import {
+  matchingActions, 
+  matchingState, 
+  matchingMutations
+} from "./store/matching.js"
```
Comment (therealyingtong) [inline frontend/src/store.js]:
```suggestion
} from "./store/matching.js";
import {
```

File: frontend/src/store.js [hunk 4]
```diff
@@ -43,20 +47,101 @@ export default new Vuex.Store({
     ...nativeLanguageState,
     ...teacherState,
     ...screeningState,
+    ...matchingState
```
Comment (therealyingtong) [inline frontend/src/store.js]:
```suggestion
    ...matchingState,
```

File: frontend/src/store.js [hunk 5]
```diff
@@ -43,20 +47,101 @@ export default new Vuex.Store({
...
     ...reasonState
   },
   mutations: {
     ...studentMutations,
     ...nativeLanguageMutations,
     ...teacherMutations,
     ...screeningMutations,
+    ...matchingMutations
```
Comment (therealyingtong) [inline frontend/src/store.js]:
```suggestion
    ...matchingMutations,
```

File: frontend/src/store.js [hunk 6]
```diff
@@ -43,20 +47,101 @@ export default new Vuex.Store({
...
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              console.log(response);
+              return;
+            }
+            dispatch('getAllStudents')
+          }
+        )
+    },
+
```
Comment (therealyingtong) [inline frontend/src/store.js]: The `updateStudentStatus()` and `updateStudentEnglishProficiency()` actions are already present in `frontend/src/store/students.js` under `studentActions`. (Do verify this for yourself.)

```suggestion
```

**Review by therealyingtong — CHANGES_REQUESTED** [review]

**Review by therealyingtong — APPROVED** [review]

**Pull request merged**